packages/main/src/plugin/kubernetes/contexts-states-dispatcher.ts

```diff
@@ -0,0 +1,34 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ContextHealthState } from './context-health-checker.js';
+import type { DispatcherEvent } from './contexts-dispatcher.js';
+import type { ContextsManagerExperimental } from './contexts-manager-experimental.js';
+
+export class ContextsStatesDispatcher {
+  constructor(private manager: ContextsManagerExperimental) {}
+
+  init(): void {
+    this.manager.onContextHealthStateChange((_state: ContextHealthState) => this.update());
+    this.manager.onContextDelete((_state: DispatcherEvent) => this.update());
+  }
+
+  update(): void {
+    console.log('current health check states', this.manager.getHealthCheckersStates());
```

Comment:
Not sure if we want to merge with this, but it can be useful for testing/reviewing, and I can remove it when LGTM'd

Reply:
ok for me to keep this trace